mono/mini/mini-profiler.c

```diff
@@ -260,12 +289,7 @@ mini_profiler_context_get_this (MonoProfilerCallContext *ctx)
 	if (ctx->interp_frame)
 		return memdup_with_type (mini_get_interp_callbacks ()->frame_get_this (ctx->interp_frame), m_class_get_this_arg (ctx->method->klass));
 
-	MonoDebugMethodJitInfo *info = mono_debug_find_method (ctx->method, mono_domain_get ());
-
-	if (!info)
-		return NULL;
-
-	return get_variable_buffer (info, info->this_var, &ctx->context);
+	return ctx->args [0];
```

Comment:
This has to use `memdup_with_type` (as in the interpreter case just above) as the returned buffer will be explicitly freed by users of `mono_profiler_call_context_get_this`.

See the API docs here: https://github.com/mono/mono/blob/master/mono/metadata/profiler.c#L661